components/scream/src/physics/shoc/shoc_compute_tmpi_impl.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef SHOC_COMPUTE_TMPI_IMPL_HPP
+#define SHOC_COMPUTE_TMPI_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::compute_tmpi(
+  const MemberType&            team,
+  const Int&                   nlevi,
+  const Scalar&                dtime,
+  const uview_1d<const Spack>& rho_zi,
+  const uview_1d<const Spack>& dz_zi,
+  const uview_1d<Spack>&       tmpi)
+{
+  const auto ggr = C::gravit;
+
+  tmpi(0)[0] = 0;
+
+  const Int nlev_pack = ekat::npack<Spack>(nlevi);
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev_pack), [&] (const Int& k) {
+    auto range_pack = ekat::range<IntSmallPack>(k*Spack::n);
+    tmpi(k).set(range_pack > 0, dtime*(ggr*rho_zi(k))/dz_zi(k));
```

Comment:
@tcclevenger , sorry, I'm just now looking at your PR. I'm confused by line 27: is there any chance for range_pack to fall below zero? If there is, I'm not seeing it.

Reply:
I think here we just want to exclude the first entry since the original for-loop in Fortran is k=2..nlev.

Reply:
Ah, ok